fsharp-backend/src/BackendOnlyStdLib/LibDarkInternal.fs

```diff
@@ -1170,4 +1170,27 @@ human-readable data."
           | _ -> incorrectArgs ())
       sqlSpec = NotQueryable
       previewable = Impure
+      deprecated = NotDeprecated }
+
+
+    { name = fn "DarkInternal" "getOpsForToplevel" 0
+      parameters = [ Param.make "canvasID" TUuid ""; Param.make "tlid" TInt "" ]
+      returnType = TList TStr
+      description = "Returns all ops for a tlid in the given canvas"
+      fn =
+        internalFn (function
+          | _, [ DUuid canvasID; DInt tlid ] ->
+            uply {
+              let tlid = uint64 tlid
+              let! ops =
+                let loadAmount = Serialize.LoadAmount.IncludeDeletedToplevels
+                Serialize.loadOplists loadAmount canvasID [ tlid ]
+
+              match ops with
+              | [ (_tlid, ops) ] -> return ops |> List.map (string >> DStr) |> DList
+              | _ -> return DList []
```

Comment:
`Serialize.loadOplists` does not return a `(tlid, ops)` pair for TLIDs that either don't exist or have no ops.

This (returning a `DList []` in such a case) seemed like a reasonable response to such, but maybe not?

Alternatively, we could update `Serialize.loadOplists` to return a pair regardless of whether it has ops or not.

Reply:
draft refactor of `Serialize.loadOplists` (just the end)

```
  Sql.query query
  |> Sql.parameters [ "canvasID", Sql.uuid canvasID; "tlids", Sql.idArray tlids ]
  |> Sql.executeAsync (fun read -> (read.tlid "tlid", read.bytea "oplist"))
  |> Task.bind (fun list ->
    tlids
    |> Task.mapWithConcurrency 2 (fun tlid ->
      match List.tryFind (fun (t, _ops) -> tlid = t) list with
      | Some (_tlid, oplist) ->
        task { return (tlid, BinarySerialization.deserializeOplist tlid oplist) }
      | None -> task { return (tlid, []) }))
```

Reply:
I would make the change in darkinternal. Empty list seems fine.